subsys/spm/spm.c

```diff
@@ -252,7 +254,7 @@ static bool usel_or_split(uint8_t id)
 	return present && (usel || split);
 }
 
-static int spm_config_peripheral(uint8_t id, bool dma_present)
+static int spm_config_peripheral(uint8_t id, bool dma_present, bool lock)
```

Comment:
I don't like this change, since it opens up for customers accidentally not locking a peripheral. AFAICS the only reason not to lock a configuration is due to debugging like we do in this PR. I would prefer if we add a separate function, which is only compiled in when certain configurations are set (such as the one in this PR). This way it is clear that not locking your peripheral is something that is ONLY done in development.

Reply:
Alternatively we could have a separate function for configuring the UART as it can be considered a special case. I just don't want to expose the option of not locking the configuration here.

Reply:
> it opens up for customers accidentally not locking a peripheral. 

It's not my area so i will end up doing what security guys are telling but my opinion first:
If a customer modifies SPM he should know what is doing. Parameter name is quite obvious and peripheral ownership can only be changed from secure state so no secure hole here as well.

Reply:
reverted that. Added dedicated `spm_config_unlocked_peripheral()` which is used for uarte0 only.